index.js

```diff
@@ -38,20 +38,22 @@ function paginate (filePath, collection, fileName, files, iteratee) {
         const posts = postsByYear[year];
         const cloneName = `${baseName}-${year}${ext}`;
 
-        const clone = cloneDeepWith(origFile, (value) => {
+        const currentYear = cloneDeepWith(origFile, (value) => {
             if (Buffer.isBuffer(value)) {
                 return value.slice();
             }
         });
 
-        last.pagination.next = clone;
-        clone.pagination.year = year;
-        clone.pagination.prev = last;
-        clone.pagination.posts = posts.map(iteratee);
```

Comment:
Previously, we mutated the `.pagination` object which was a copy of last year's pagination object -- that included the `.pagination.next` property.

Now that we're always re-creating a new `.pagination` object, we're ensuring that the last year's `.pagination.next` isn't present anymore.

That property is only set in the _next iteration_, which doesn't happen for the oldest year -- voila, bug fixed.